src/Imath/half.h

```diff
@@ -17,33 +17,149 @@
 ///
-/// Further, Additional control is possible for embedded
+/// **Representation of a 32-bit float:**
+///
+/// We assume that a float, f, is an IEEE 754 single-precision
+/// floating point number, whose bits are arranged as follows:
+///
+///     31 (msb)
+///     |
+///     | 30     23
+///     | |      |
+///     | |      | 22                    0 (lsb)
+///     | |      | |                     |
+///     X XXXXXXXX XXXXXXXXXXXXXXXXXXXXXXX
+///
+///     s e        m
+///
+/// S is the sign-bit, e is the exponent and m is the significand.
+///
+/// If e is between 1 and 254, f is a normalized number:
+///
+///             s    e-127
+///     f = (-1)  * 2      * 1.m
+///
+/// If e is 0, and m is not zero, f is a denormalized number:
+///
+///             s    -126
+///     f = (-1)  * 2      * 0.m
+///
+/// If e and m are both zero, f is zero:
+///
+///     f = 0.0
+///
+/// If e is 255, f is an "infinity" or "not a number" (NAN),
+/// depending on whether m is zero or not.
+///
+/// Examples:
+///
+///     0 00000000 00000000000000000000000 = 0.0
+///     0 01111110 00000000000000000000000 = 0.5
+///     0 01111111 00000000000000000000000 = 1.0
+///     0 10000000 00000000000000000000000 = 2.0
+///     0 10000000 10000000000000000000000 = 3.0
+///     1 10000101 11110000010000000000000 = -124.0625
+///     0 11111111 00000000000000000000000 = +infinity
+///     1 11111111 00000000000000000000000 = -infinity
+///     0 11111111 10000000000000000000000 = NAN
+///     1 11111111 11111111111111111111111 = NAN
+///
+/// **Representation of a 16-bit half:**
+///
+/// Here is the bit-layout for a half number, h:
+///
+///     15 (msb)
+///     |
+///     | 14  10
+///     | |   |
+///     | |   | 9        0 (lsb)
+///     | |   | |        |
+///     X XXXXX XXXXXXXXXX
+///
+///     s e     m
+///
+/// S is the sign-bit, e is the exponent and m is the significand.
+///
+/// If e is between 1 and 30, h is a normalized number:
+///
+///             s    e-15
+///     h = (-1)  * 2     * 1.m
+///
+/// If e is 0, and m is not zero, h is a denormalized number:
+///
+///             S    -14
+///     h = (-1)  * 2     * 0.m
+///
+/// If e and m are both zero, h is zero:
+///
+///     h = 0.0
+///
+/// If e is 31, h is an "infinity" or "not a number" (NAN),
+/// depending on whether m is zero or not.
+///
+/// Examples:
+///
+///     0 00000 0000000000 = 0.0
+///     0 01110 0000000000 = 0.5
+///     0 01111 0000000000 = 1.0
+///     0 10000 0000000000 = 2.0
+///     0 10000 1000000000 = 3.0
+///     1 10101 1111000001 = -124.0625
+///     0 11111 0000000000 = +infinity
+///     1 11111 0000000000 = -infinity
+///     0 11111 1000000000 = NAN
+///     1 11111 1111111111 = NAN
+///
+/// **Conversion via Lookup Table:**
+///
+/// Converting from a float to a half requires some non-trivial bit
+/// manipulations.  In some cases, this makes conversion relatively
+/// slow, but the most common case is accelerated via table lookups.
```

Comment:
We should probably change this. I might suggest something like:

Converting from a float to a half requires some non-trivial bit manipulations, and quite commonly has to deal with denormalized numbers, and has to always handle rounding. The rounding is always round to even. If possible on modern hardware, it is suggested to use the hardware intrinsics.